app/presenters/keyset_pagination_presenter.rb

```diff
@@ -36,27 +36,36 @@ def next_link
     end
 
     def self_link
-      { href: request_url, rel: "self" }
+      { href: self_url, rel: "self" }
     end
 
     def previous_link
       { href: previous_url, rel: "previous" } unless pagination_query.is_first_page?
     end
 
     def next_url
-      page_href(after: pagination_query.next_after_key.join(","))
+      page_href(except: %w(before after), after: pagination_query.next_after_key.join(","))
     end
 
     def previous_url
-      page_href(before: pagination_query.next_before_key.join(","))
+      page_href(except: %w(before after), before: pagination_query.next_before_key.join(","))
     end
 
-    def page_href(params)
+    def self_url
+      page_href
+    end
+
+    def page_href(params = {})
+      except = params.delete(:except)
```

Comment:
Rather than needing to pass an except in should we just remove them if there is a before or after in the params?